packages/backend-for-frontend/src/services/clientService.ts

```diff
@@ -249,6 +250,20 @@ export function clientServiceBuilder(
           )
         )
       );
+
+      /*      
+       1) name.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') escapes all special regex characters.
+       2) The 'i' flag makes the match case-insensitive.
+       3) .test(user.name) tests whether user.name contains the string name in part.
+      */
+
+      return name
+        ? users.filter((user) =>
+            new RegExp(name.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"), "i").test(
```

Comment:
```suggestion

      return name
        ? users.filter((user) =>
            new RegExp(ReadModelRepository.escapeRegExp(name), "i").test(
```
You can reuse this already defined function